classes/order/OrderHistory.php

```diff
@@ -615,4 +627,47 @@ public function addWs()
             return $this->add();
         }
     }
+
+    /**
+     * @return MailPartialTemplateRenderer
+     */
+    protected function getPartialRenderer()
+    {
+        if (!$this->partialRenderer) {
+            $this->partialRenderer = new MailPartialTemplateRenderer($this->context->smarty);
+        }
+
+        return $this->partialRenderer;
+    }
+
+    /**
+     * Fetch the content of $template_name inside the folder
+     * current_theme/mails/current_iso_lang/ if found, otherwise in
+     * mails/current_iso_lang.
+     *
+     * @param string $template_name template name with extension
+     * @param int $mail_type Mail::TYPE_HTML or Mail::TYPE_TEXT
+     * @param array $var sent to smarty as 'list'
+     *
+     * @return string
+     */
+    protected function getEmailTemplateContent($template_name, $mail_type, $var)
```

Comment:
As I told you I don't think this extra function is useful, it was already present in PaymentModule so I just modified the last part about partiel rendering. But anyway in the Mail::send function we already check the PS_MAIL_TYPE configuration and this is where the appropriate content is selected (or not) inside the mail body.

So I think you could simply (even drop completely) this method and simply call `$this->getPartialRenderer()->render($template_name, $this->context->language, $var)` to create you two variables
Sure in cas the mail config is not for BOTH you will have generated one for nothin but I the performance cost is really minimal compared to the complexity of the code

Reply:
Actually it's probably useless in PaymentModule as well, but I wanted to make the minimum changes possible I guess